docs/test-plan/verification-tp.md

```diff
@@ -0,0 +1,31 @@
+# Verification Test Plan
+
+## Overview
+
+Verification is one of the most important functions of a BlockNode, as it is responsible for ensuring the integrity of the blockstream and that the data is correct—as reached by consensus across the network. This document outlines the test plan for the verification module, detailing both positive and negative test cases for the available implementations.
+
+## Key Considerations
+
+- **Implementation Variants:** Two implementations exist, SYNC for testing and ASYNC for production.
+
+- **ASYNC vs. SYNC:** The ASYNC implementation uses a thread pool with a `concurrent` TreeHasher for calculating the block root hash and is approximately 50% faster than the SYNC version, which uses a `naive` TreeHash.
+
+- **Naive Tree Hasher:** calculates the root hash of the merkle trees in a synchronous manner and using a simple recursive algorithm.
+
+- **Concurrent Tree Hasher:** calculates the root hash of the merkle trees in an asynchronous manner using a thread pool to parallelize the calculation as the blocks items are received and leafs are appended to the "merkle tree".
```

Comment:
Q: Is it important to note the logic of parallelization here in case it informs testing approach or is that an implementation not necessary?